subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/ParallelDownloadsIntegrationTest.groovy

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.resolve
+
+import org.gradle.integtests.fixtures.AbstractHttpDependencyResolutionTest
+import spock.lang.Unroll
+
+import javax.servlet.http.HttpServletRequest
+import java.util.concurrent.CyclicBarrier
+import java.util.concurrent.TimeUnit
+
+class ParallelDownloadsIntegrationTest extends AbstractHttpDependencyResolutionTest {
```

Comment:
Yes, that's how I noticed it didn't work with composite builds: a test is failing, so already covered. As for substituted dependencies, I could add a test but it doesn't make sense, since it happens at a different time.